pubnub-kotlin/pubnub-kotlin-impl/src/main/kotlin/com/pubnub/internal/PubNubImpl.kt

```diff
@@ -179,6 +179,11 @@ open class PubNubImpl(
     val pnsdkName: String = PNSDK_PUBNUB_KOTLIN,
     eventEnginesConf: EventEnginesConf = EventEnginesConf()
 ) : PubNub {
+    internal val tokenManager: TokenManager = TokenManager()
+
+    init {
+        this.setToken(configuration.authToken)
```

Comment:
this will overwrite the `null` value in TokenManager with the default of `""` (empty string) from PNConfiguration, which will stop `authKey` (the deprecated one) from ever being used (because the order in `EndpointCore` is `tokenManager.getToken()`, then `authKey`, and `tokenManager.getToken` will now never be `null`)

I think we should make `configuration.authToken` nullable and null by default (instead of empty string) which I think will solve this. It will be different then other keys (which use empty strings), but IMHO we should over time move all keys to `String? = null` instead of `String = ""`, it's just not right to use empty strings there.

Reply:
Good catch. I will change the code.